pyslim/slim_metadata.py

```diff
@@ -11,6 +13,431 @@
+                     "slim_time" : 243,
+                     "nucleotide" : 3
+                 } ]
+                }
+            ]
+        },
+        "node" : 
+        {
+            "$schema": "http://json-schema.org/schema#",
+            "description": "SLiM schema for node metadata.",
+            "codec": "struct",
+            "type": ["object", "null"],
+            "properties": {
+                "slim_id": {
+                    "type": "integer",
+                    "description": "The 'pedigree ID' of this chromosome in SLiM.",
+                    "binaryFormat": "q",
+                    "index": 0},
+                "is_null": {
+                    "type": "integer",
+                    "description": "Whether this node describes a 'null' (non-existant) chromosome.",
+                    "binaryFormat": "B",
+                    "index": 1},
+                "genome_type": {
+                    "type": "integer",
+                    "description": "The 'type' of this genome (0 for autosome, 1 for X, 2 for Y).",
+                    "binaryFormat":
+                    "B", "index": 2}
+            },
+            "required": ["slim_id", "is_null", "genome_type"],
+            "additionalProperties": False,
+            "examples": [
+                {
+                "slim_id" : 123,
+                "is_null" : False,
+                "genome_type" : 0
+                }
+            ]
+        },
+        "individual" :
+        {
+            "$schema": "http://json-schema.org/schema#",
+            "description": "SLiM schema for individual metadata.",
+            "codec": "struct",
+            "type": "object",
+            "properties": {
+                "pedigree_id": {
+                    "type": "integer",
+                    "description": "The 'pedigree ID' of this individual in SLiM.",
+                    "binaryFormat": "q",
+                    "index": 1
+                },
+                "age": {
+                    "type": "integer",
+                    "description": "The age of this individual, either when the tree sequence was written out (if the individual was alive then), or the last time they were Remembered (if not).",
+                    "binaryFormat": "i",
+                    "index": 2
+                },
+                "subpopulation": {
+                    "type": "integer",
+                    "description": "The ID of the subpopulation the individual was part of, either when the tree sequence was written out (if the individual was alive then), or the last time they were Remembered (if not).",
+                    "binaryFormat": "i",
+                    "index": 3
+                },
+                "sex": {
+                    "type": "integer",
+                    "description": "The sex of the individual (0 for female, 1 for male, -1 for hermaphrodite).",
+                    "binaryFormat": "i",
+                    "index": 4
+                },
```

Comment:
This could also have `"enum": [-1,0,-1]` added.

Reply:
Right - but, some arguments against: (a) we'd be causing ourselves future headaches if we wanted to add to the list; and (b) perhaps it would slow down decoding? But I haven't tested to see how much it would.

Reply:
Would slightly slow down validation - but not encoding or decoding which will ignore it.

Reply:
Hm. I think we'll omit it, since it's documented here, which seems good enough, to avoid future headaches.